homeassistant/components/adguard/config_flow.py

```diff
@@ -15,17 +18,20 @@
 )
 from homeassistant.helpers.aiohttp_client import async_get_clientsession
 
+from .const import DOMAIN  # pylint: disable=unused-import
+
 
-@config_entries.HANDLERS.register(DOMAIN)
-class AdGuardHomeFlowHandler(ConfigFlow):
+class AdGuardHomeFlowHandler(ConfigFlow, domain=DOMAIN):
     """Handle a AdGuard Home config flow."""
 
     VERSION = 1
     CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
 
     _hassio_discovery = None
 
-    async def _show_setup_form(self, errors=None):
+    async def _show_setup_form(
+        self, errors: dict[str, str] | None = None
```

Comment:
Aren't union types using the pipe (`|`)  notation new in python 3.10? See https://docs.python.org/3.10/library/typing.html#typing.Union.

Reply:
Future import